doc/nrf/ug_zigbee_configuring_zboss_traces.rst

```diff
@@ -0,0 +1,187 @@
+     These libraries are not certified and should not be used for production firmware or end products.
+
+
+Requirements
+************
+
+To collect and view the log files, you will need the following:
+
+* Software that allows for reading serial ports.
+  For example, if you are using Windows you can use :ref:`putty`.
+
+* Correct COM port.
+
+   * For traces using the UART, the JLink COM port is used. The development kit is assigned to a COM port (Windows) or a ttyACM device (Linux), which is visible in the system's :guilabel:`Device Manager`.
+
+   * For traces using USB, a virtual COM port (a serial port emulated over USB) is used. You can set :kconfig:`CONFIG_USB_DEVICE_PRODUCT` to help identify the COM port in the system's :guilabel:`Device Manager`.
+
+* An additional USB cable if trace logs through USB are enabled (see :ref:`ug_zigbee_configuring_zboss_traces_using_usb`).
+
+
+.. rst-class:: numbered-step
+
+Switch to ZBOSS libraries with compiled-in trace logs
+*****************************************************
+
+Set the Kconfig option :kconfig:`CONFIG_ZIGBEE_ENABLE_TRACES` to switch to ZBOSS libraries with compiled-in trace logs.
+
+The ZBOSS stack comes in a precompiled form and trace logs are not compiled-in by default.
+An additional set of ZBOSS libraries are available in nrfxlib, which does have trace logs compiled-in.
+
+
+.. rst-class:: numbered-step
+
+Select which ZBOSS trace logs to print
+**************************************
+
+Complete the following steps:
+
+1. Select from which subsystems you would like to receive logs by configuring the ZBOSS trace mask with the Kconfig option :kconfig:`CONFIG_ZBOSS_TRACE_MASK`.
+   Trace masks can be created by adding up masks of subsystems to receive the trace logs from.
+   For available subsystems, see :file:`nrfxlib/zboss/production/include/zb_trace.h`.
+
+#. Select the level of logs you want to receive.
+   Configure ZBOSS trace level by selecting one of the following levels:
+
+   * Error trace logs level - set :kconfig:`CONFIG_ZBOSS_TRACE_LOG_LEVEL_ERR`
+   * Warning trace logs level - set :kconfig:`CONFIG_ZBOSS_TRACE_LOG_LEVEL_WRN`
+   * Info trace logs level - set :kconfig:`CONFIG_ZBOSS_TRACE_LOG_LEVEL_INF`
+   * Debug trace logs level - set :kconfig:`CONFIG_ZBOSS_TRACE_LOG_LEVEL_DBG`
+
+If you do not want to receive trace logs, turn them off by setting the Kconfig option :kconfig:`CONFIG_ZBOSS_TRACE_LOG_LEVEL_OFF`.
+
+Each of the following levels on the list also includes the previous one.
+See :ref:`zigbee_ug_logging_stack_logs` to read more about trace logs.
+
+
+.. rst-class:: numbered-step
+
+Configure how to print ZBOSS trace logs
+***************************************
+
+The :ref:`zigbee_osif_zboss_osif_serial` offers a few backends to choose from for printing ZBOSS trace logs.
+It is recommended to use the Zigbee serial logger, as it is the most efficient.
+To enable it, set the Kconfig option :kconfig:`CONFIG_ZBOSS_TRACE_BINARY_LOGGING`.
+
+Optional
```

Comment:
```suggestion
Optional: Increasing ring buffer size
```
I think having more information in this line will help the user notice what this is about.

Reply:
Hm, I think this can work, yes, but the link you provide has less info about the ring buffer than we have here.
So maybe:
```suggestion
Optional: Increasing ring buffer size

   You can increase size of the ring buffer that temporarily stores the trace logs.
   To do this, use :kconfig:`CONFIG_ZBOSS_TRACE_LOGGER_BUFFER_SIZE` to assign a value for size of the buffer.
   See :ref:`Zigbee serial logger <zigbee_osif_zigbee_async_serial>` for more information.
```
And add this info to the linked section, next to "Data is buffered internally in ring buffer and printed in the binary format.":
```
This ring buffer has a size of 4096 bytes by default.
This can prevent losing some of the logs in demanding scenarios such as high network traffic, multiple devices being configured or joined, and so on.
```

Reply:
@sebastiandraus , any thoughts? :)

Reply:
`This can prevent losing some of the logs in demanding scenarios such as high network traffic, multiple devices being configured or joined, and so on.`
It think that this is not necessary in the linked section.

Reply:
Okay, will only add: ```This ring buffer has a size of 4096 bytes by default.``` to the ZBOSS OSIF doc.